### PR TITLE
Stats

### PR DESCRIPTION
**Pull request by justinlittman**
## Why was this change made?
To provide some basic information on technical metadata processing.

## Was the usage documentation (e.g. openapi.yml, README, DevOpsDocs) updated?
No.

## Does this change affect how this application integrates with other services?
No.

### CODE REVIEW

**Comment by jcoyne** [comment]
👎 to putting this in.  I'd much prefer to see this via an API and have Argo call it.
- Don't have a js/bootstrap dependency
- Don't need special handling for openapi routes/vs UI routes.
- Don't have to consider authorization (via Shibboleth)
- This is not a very discoverable place to put something, Argo might serve the audience better and benefit more people.

**Comment by justinlittman** [comment]
My case for it:
* We already have a monitoring UI in most API-only apps, viz., Sidekiq UI and already need to provide special handling. This isn't that burdensome.
* Adding dependencies to allow a Rails app to have a web UI shouldn't be considered burdensome. (I'd actually suggest the opposite; all of our Rails apps should be configured for web UIs, as having some apps that do and some that don't is more confusing than having all the same. We'd be helped by configuration consistency rather than the current inconsistency, especially for those of us who don't understand the finer points of Rails configuration.)
* This is intended to be quick and dirty monitoring/stats for our and Andrew's use. If we want to expose something for other users then by all means should consider adding to Argo.
* In the spirit of separate microservices, just as each application is responsible for it's own datastore and provides it's own status, it should also provide it's own monitoring / stats.
* API applications as black boxes has been problematic for us. I'd suggest that one of the reasons that errors in pres cat have received such little attention is because to understand what is going on you have to look at the logs / query the DB directly. Exposing these sorts of stats in a simple way allows us to better understand if a service is operating properly. Similarly, for other API applications such as the SDR API, I'd suggest that the lack of Honeybadger errors and that status checks pass isn't really very good evidence that the application is actually functioning properly.
* Exposing this as an API and consuming it in Argo is way more complicated than some simple Rails template rendering. Adding a new stat would require changes in the view, the OpenAPI, and then in Argo, making it less likely that one would bother. (And that's ignoring if we end up having a client.)

**Comment by jcoyne** [comment]
I would go so far as to say that any server that provides an API and a web UI is by definition not a microservice.

**Comment by aaron-collier** [comment]
Hmmmm... I'm leaning towards @jcoyne's point on this. While we did put a simple UI on google-books, the basic data there may be better served through API endpoint(s) and surfaced in a central UI like argo. I can see a better stats dashboard available there than to have to go to individual endpoints.

**Comment by peetucket** [comment]
Personally I would take the ability to immediately see the stats even though it incurs a tradeoff of increased complexity in the app.  I could imagine a future world with a dashboard (maybe in Argo) that fetches the stats from each ancillary service for easier consumption in a single location, but thats unlikely to be anytime soon unfortunately.  It also seems that doing that would minimally require an API that exposes those stats anyway (i.e. controllers and other bits of rails), so adding views in the app itself doesn't feel that bad.

**Comment by jcoyne** [comment]
@peetucket who should be able to see this and who shouldn't?

**Comment by justinlittman** [comment]
Who should be able to see sidekiq ui and who shouldn't?

(My opinion: I don't think it matters.)

**Comment by peetucket** [comment]
> @peetucket who should be able to see this and who shouldn't?

That's a good question and not sure its my call to make.  For this particular service, I'd imagine infra developers (and Andrew if he's not in that group) would be a good starting point.

**Comment by jcoyne** [comment]
@andrewjbtw who should be able to access this page?

**Comment by andrewjbtw** [comment]
I don't think it needs to be accessible beyond developers plus the repo manager at first. But I still haven't been able to reach the stats page myself.

File: app/controllers/stats_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+# StatsController surfaces specific stat values for display
+class StatsController < ApplicationController
```
Comment (jcoyne) [inline app/controllers/stats_controller.rb]: Add Shibboleth authorization

Comment (justinlittman) [reply app/controllers/stats_controller.rb]: https://github.com/sul-dlss/operations-tasks/issues/2184

**Comment by justinlittman** [comment]
@aaron-collier Added mention to README: https://github.com/sul-dlss/technical-metadata-service/pull/91/files#diff-04c6e90faac2675aa89e2176d2eec7d8R24-R27

**Pull request merged**